Doc/library/typing.rst

```diff
@@ -3235,6 +3235,35 @@ Introspection helpers
 
    .. versionadded:: 3.7.4
 
+   .. versionchanged:: 3.14
+      This is now an alias for :class:`annotationlib.ForwardRef`.
+
+.. function:: evaluate_forward_ref(forward_ref, *, owner=None, globals=None, locals=None, type_params=None, format=annotationlib.Format.VALUE)
+
+   Evaluate an :class:`annotationlib.ForwardRef` as a :term:`type hint`.
+
+   This is similar to calling the :meth:`annotationlib.ForwardRef.evaluate()` method,
+   but unlike that method, :func:`!evaluate_forward_ref` also:
+
+   * Recursively evaluates forward references nested within the type hint.
+   * Rejects certain objects that are not valid type hints.
+   * Replaces type hints that evaluate to None with types.NoneType.
+   * Supports the :attr:`~annotationlib.Format.FORWARDREF` and
+     :attr:`~annotationlib.Format.SOURCE` formats.
+
+   *forward_ref* must be an instance of :class:`~annotationlib.ForwardRef`. *owner*, if given,
+   should be the object that holds the annotations that the forward reference
+   derived from, such as a module, class object, or function. It is used to
+   infer the namespaces to use for looking up names. *globals* and *locals*
+   can also be explicitly given to provide the global and local namespaces.
+   *type_params* is a tuple of :ref:`type parameters <type-params>` that are in scope when
+   evaluating the forward reference. This parameter must be provided (though
+   it may be an empty tuple) if *owner* is not given and the forward reference
+   does not already have an owner set. *format* specifies the format of the
+   annotation and is a member of the :class:`annotationlib.Format` enum.
```

Comment:
Reflow for line length
```suggestion
   *forward_ref* must be an instance of :class:`~annotationlib.ForwardRef`.
   *owner*, if given, should be the object that holds the annotations that
   the forward reference derived from, such as a module, class object, or function.
   It is used to infer the namespaces to use for looking up names. 
   *globals* and *locals* can also be explicitly given to provide 
   the global and local namespaces.
   *type_params* is a tuple of :ref:`type parameters <type-params>` that
   are in scope when evaluating the forward reference.
   This parameter must be provided (though it may be an empty tuple) if *owner*
   is not given and the forward reference does not already have an owner set.
   *format* specifies the format of the annotation and is a member of
   the :class:`annotationlib.Format` enum.
```